### PR TITLE
Policies, TargetPolicies, Reports and Jobs APIs 

### PR DESCRIPTION
**Pull request by walker2**
# Description
- Improved and extended `policies` API 
- Added `targetPolicies`, `reports` and `jobs` APIs 
- Added replication-related calls such as `SyncPolicy`, `AllowWrites` etc.
- Added custom waiting/polling logic 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/116  |

# Common PR Checklist:

- [x] Have you made sure that the code compiles?
- [x] Have you commented your code, particularly in hard-to-understand areas
- [x] Did you run tests in a real Kubernetes cluster?
- [x] Have you maintained backward compatibility

### CODE REVIEW

File: acls.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* 
+/*
  Copyright (c) 2019 Dell Inc, or its subsidiaries.
```
Comment (AronAtDell) [inline acls.go:2]: As you are in this folder. Please adjust the copyright year to reflect this year. Thanks!

File: api/v11/api_v11_replication.go [hunk 2]
```diff
@@ -41,23 +135,98 @@ func GetPolicyByName(
...
+func StartSyncIQJob(ctx context.Context, client api.Client, job *JobRequest) (*Job, error) {
+	var jobResp Job
+	return &jobResp, client.Post(ctx, jobsPath, "", nil, nil, job, &jobResp)
+}
+
+func GetReport(ctx context.Context, client api.Client, reportName string) (*Report, error) {
+	r := &Reports{}
+	err := client.Get(ctx, reportsPath, reportName, nil, nil, &r)
+	if err != nil || len(r.Reports) == 0 {
+		return nil, err
```
Comment (randeepsharma) [inline api/v11/api_v11_replication.go:195]: Recommend to add change similar to the one in line 213 as in the second check err is nil.
if len(r.Reports) == 0 {
		return nil, fmt.Errorf("no reports found for policy %s", policyName)
	}

**Review by randeepsharma — APPROVED** [review]

File: api/v11/api_v11_replication.go (hunk 2 above)
Comment (walker2) [reply api/v11/api_v11_replication.go:210]: Done

File: acls.go (hunk 1 above)
Comment (walker2) [reply acls.go]: Updated and added to newly created files

**Review by maxan98 — APPROVED** [review]

**Review by IliaRN — APPROVED** [review]

**Review by bpjain2004 — APPROVED** [review]

**Pull request merged**